dsl/src/main/groovy/org/cristalise/dsl/persistency/outcome/SchemaDelegate.groovy

```diff
@@ -155,16 +169,13 @@ class SchemaDelegate {
         }
     }
 
-    private String fieldType(Field f) {
-        if (!f.values && !f.unit && !f.pattern && !hasRangeConstraints(f) && !f.attributes)
-            return f.type
-        else
-            return ''
-    }
-
     private void buildField(xsd, Field f) {
         Logger.msg 1, "SchemaDelegate.buildField() - Field: $f.name"
 
+        //TODO: implement support for this combination - see issue 129
+        if (f.attributes && f.unit && hasRestrictions(f)) 
```

Comment:
all three have to be present? I think it should be `((f.attributes || f.unit) && hasRestrictions(f))`